docs/guide/mocking.md

```diff
@@ -602,12 +602,17 @@ class Dog {
 
   constructor(name: string) {
     this.name = name
+    this.greet = this.greet.bind(this)
```

Comment:
Why do we need this in the example? Doesn't it already work like this?

Reply:
I am not sure why the whole line is needed. The method is already defined as a public class method

Reply:
Basically, my question is why can't we just have this:

```ts
class Dog {
  name: string

  constructor(name: string) {
    this.name = name
  }

  static getType(): string {
    return 'animal'
  }

  greet(): string {
    return `Hi! My name is ${this.name}!`
  }

  speak(): string {
    return 'bark!'
  }

  isHungry() {}
  feed() {}
}
```

Reply:
> Why do we need to repeat it in the constructor? This code is how the source code looks like - people usually don't duplicate this

I'm a bit biased for sure, binding class methods like that was pretty common in React when classes was the default API to use React.

With the example, I want to simulate a method that only exists in the object instance.
Would it be clearer if I changed to a class field with arrow function like mentioned above:
```
  greet = () => {
    return `Hi! My name is ${this.name}!`
  }
```
would be equivalent to
```
    constructor(name) {
        this.greet = () => {
            return `Hi! My name is ${this.name}!`;
        };
    }
```

Reply:
Sure, this makes more sense to me in this example

Reply:
Awesome, makes sense, thanks for the thorough review 👍 

Finished updating based on your comment.